packages/vm/src/types.ts

```diff
@@ -140,6 +140,12 @@ export interface VMOpts {
    * Use a custom EVM to run Messages on. If this is not present, use the default EVM.
    */
   evm?: EVMInterface
+
+  profilerOpts?: {
+    //evmProfilerOpts: EVMProfilerOpts
+    reportProfilerAfterTx?: boolean
+    reportProfilerAfterBlock?: boolean
```

Comment:
Two quick nits: I would drop the `Profiler` here from the two options, so just `reportAfterTx` and `reportAfterBlock`, context is already clear and option gets shorter.

Reply:
I have explicitly added this profiler to VM, because we want to (?) profile VM pre-setup-before EVM call and post-setup-after-EVM call too. So after the EVM is ran, then VM will flush the StateManager and this can take some time in some situations, which we want to profile to (?)

Reply:
Ah you are right, will update

Reply:
I would actually still be a fan of this name changes (can be along some follow-up work), so that we get to e.g. `this._opts.profilerOpts?.reportAfterBlock` (still long enough 🙂).

Reply:
> I would actually still be a fan of this name changes (can be along some follow-up work), so that we get to e.g. `this._opts.profilerOpts?.reportAfterBlock` (still long enough 🙂).

Yes, added this :)